lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -51,6 +58,48 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+TF_DEFINE_PRIVATE_TOKENS(_tokens, (inputs)(varname));
+
+namespace {
+// We want to know if this material is a specialization that was created to handle UV mappings on
+// export. For details, see the _UVMappingManager class in ..\shading\shadingModeExporterContext.cpp
+//
+// If the root layer contains anything that is not a varname input, we do not consider it a
+// mergeable material.
+bool _IsMergeableMaterial(const UsdShadeMaterial& shadeMaterial)
+{
+    if (!shadeMaterial || !shadeMaterial.HasBaseMaterial()) {
+        return false;
+    }
+    const PcpPrimIndex&   primIndex = shadeMaterial.GetPrim().GetPrimIndex();
+    PcpNodeRef            primRoot = primIndex.GetRootNode();
+    SdfPath               primPath = primIndex.GetPath();
+    const SdfLayerHandle& layer = primRoot.GetLayerStack()->GetLayerTree()->GetLayer();
+
+    bool retVal = true;
+    auto testFunction = [&](const SdfPath& path) {
+        // If we traverse anything that is not a varname specialization, we return false.
+        if (path == primPath) {
+            return;
+        }
+        if (path.GetParentPath() != primPath || !path.IsPrimPropertyPath()) {
+            retVal = false;
+            return;
+        }
+        std::vector<std::string> splitName = SdfPath::TokenizeIdentifier(path.GetName());
+        // We allow only ["inputs", "XXX", "varname"]
+        if (splitName.size() != 3 || splitName[0] != _tokens->inputs.GetString()
+            || splitName[2] != _tokens->varname.GetString()) {
+            retVal = false;
+        }
+    };
+
+    layer->Traverse(primPath, testFunction);
```

Comment:
Along the lines of the above comment, it seems like we'd want a stage-based traversal here rather than a prim-based one.